src/docs/arch-hs.dox

```diff
@@ -1,8 +1,272 @@
 /*! @arch_page arch-hs History Store
 
-The History Store in WiredTiger tracks old (all but the latest committed)
-versions of records.  By having these records in storage separate from
-the current version, they can be used to service long running transactions,
+The history store in WiredTiger tracks historical versions of records required to
+service older readers. By having these records in storage separate from
+the current version, they can be used to service long running transactions
 and be evicted as necessary, without interfering with activity that uses
-the most recent committed versions.
+the most recent committed versions. With the introduction of history store, only the
+newest update to a key is written to the user table while the older updates
+for the key are written to the history store. All user tables in the database are
+backed by a single history store table. The history store has no outward
+visibility to the application code and only WiredTiger's internal modules can
+perform operations on the history store table using a predefined cursor API.
+
+@section arch_hs_table History store table structure
+WiredTiger writes the history store on the disk as a standard row-store table. The key for
+the history store table is a combination of:
```

Comment:
`The key for the history store table is a combination of`, I suggest we could reword this to:
`Each key in the history store is a combination of` or `The key for a given update in the history store table is a combination of`.
It is just a suggestion, it is surely correct this way but I thought we could reword to clarify.

Reply:
Every key for a particular table follows a preconfigured format. Using words "each" or "key for given" can imply that we could have used different key format for each entry. I believe the existing sentence is sufficient here.